### DISCUSSION TITLE
A possibility to use normalization parameter in ORDER BY _relevance construction

### DISCUSSION THREAD

**Discussion by oartemios** [prisma/prisma#22302] [Q&A] [answered]
### Question

Hi everyone!

Does Prisma support ORDER BY _relevance with passing extra parameter 'normalization' (default is 0, PG supports values: 1, 2, 4, 8, 16, 32) that should affect TS_RANK calculation result?

From PG documentation:
```
ts_rank([ weights float4[], ] vector tsvector, query tsquery [, normalization integer ]) returns float4
```

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @oartemios 👋 

Thank you for raising this question.

The [documentation](https://www.prisma.io/docs/concepts/components/prisma-client/filtering-and-sorting#sort-by-relevance-postgresql) only cover the basic usage of  `_relevance` for sorting by relevance in full-text search, without delving into the specifics of the `ts_rank` function or its parameters like `normalization`. You may need to use [raw query](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access) to perform the relevance sorting you desire. You can leverage [SafeQL](https://www.prisma.io/docs/guides/other/advanced-database-tasks/typesafe-raw-sql/safeql) to create type safe raw queries. 

To see support for using `normalization` parmeter in `ORDER BY _relevance` construction, you can open a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) so that our engineering team can take a look.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by oartemios** [reply]
Would you mind if we try to contribute to this issue?

**Reply by oartemios** [reply]
Hi @ludralph! 

What do you think about my suggestion?
Are there any obstacles for that?

**Reply by ludralph** [reply, marked as answer]
Hi @oartemios 👋 

Did you open a feature request for this? That's the first step to help our engineering team understand the issue and from there you could provide your contributions.

**Reply by oartemios** [reply]
Feature request is created https://github.com/prisma/prisma/issues/22390
